src/lib.rs

```diff
@@ -183,12 +222,12 @@ impl<IoPin, ActiveLevel> Switch<IoPin, ActiveLevel> {
     /// ```
     pub fn new(pin: IoPin) -> Self {
         Switch {
-            pin: pin,
+            pin,
```

Comment:
I’m aware the language allows you to shortcut if the param name matches the field name, but prefer being explicit.